src/app/modules/submission/submission.service.ts

```diff
@@ -221,43 +182,39 @@ export const getFormSubmissionsCount = (
  * @param param0 Data to include in email confirmations
  * @param param0.form Form object
  * @param param0.submission Submission object which was saved to database
- * @param param0.parsedResponses Responses for each field
- * @param param0.autoReplyData Subset of responses to be included in email confirmation
+ * @param param0.responsesData Subset of responses to be included in email confirmation
  * @param param0.attachments Attachments to be included in email
+ * @param autoReplyData Array of objects that contains autoreply mail data to override with defaults
  * @returns ok(true) if all emails were sent successfully
  * @returns err(SendEmailConfirmationError) if any email failed to be sent
  */
 export const sendEmailConfirmations = <S extends ISubmissionSchema>({
   form,
   submission,
-  parsedResponses,
-  autoReplyData,
+  responsesData = [],
   attachments,
+  autoReplyData,
 }: {
   form: IPopulatedForm
   submission: S
-  parsedResponses: ProcessedFieldResponse[]
-  autoReplyData?: EmailRespondentConfirmationField[]
+  responsesData?: EmailRespondentConfirmationField[]
   attachments?: IAttachmentInfo[]
+  autoReplyData: AutoReplyMailData[]
```

Comment:
the variable naming here has always been quite confusing. would `recipientData: AutoReplyMailData` be more reflective of the purpose of this param?

Reply:
👌 made the change